### PR TITLE
Default DApp data structure (#580)

### LINKED ISSUE #580: Default DApp data structure
During alpha and building out developer tooling we should allow DApp developers to easily submit their DApps as a PR with a name, URL, and icon, which would create a pseudo contact that opens up the !browse command - similar to Wallet, there is no need for any chat integration.

the dapps would appear in dapps group of contacts screen

ideally this just some list in a file with an icon going into resources dir.

Edit: Actually can we generalise this as well to allow adding contacts to groups, I would like to add "Jarrad" and my Status public key to People group.

### PR DESCRIPTION
**Pull request by alwx**
Fixes #580

### CODE REVIEW

**Comment by alwx** [comment]
There is a file `resources/default_contacts.json`, which looks this way:

```
[
    {"id": "wallet",
     "name": "Wallet",
     "photo-path": "icon_wallet_avatar",
     "add-chat?": true,
     "dapp?": true},

    {"id": "dapp-auction-house",
     "name": "Auction House",
     "photo-path": "http://auctionhouse.dappbench.com/images/auctionhouse.png",
     "dapp?": true,
     "dapp-url": "http://auctionhouse.dappbench.com"}
]
```

You can simply add any other DApp to this list and it will be added to contacts. This approach also allows you to add any other contact (not only DApp) with any public-key value you want.

**Comment by jarradh** [comment]
Curious if `photo-path` can also point to a file in resources? (and work on both ios/android)

**Comment by rasom** [comment]
I suppose it could be done by code like this https://github.com/status-im/status-react/blob/cb2709cef49906fc956eca8acba49b2392999fa5/src/status_im/resources.cljs#L3-L3

**Comment by alwx** [comment]
@rasom yes you're right, we need to pre-require the image in some kind of resources file before.

**Comment by jarradh** [comment]
I'm fine to do it with `status-im.resources`

**Comment by alwx** [comment]
@jarradh 

resources.cljs:
```
(def contacts
  {:auction-house (js/require "./images/contacts/auction-house.png")})
```

resources/default_contacts.json:
```
[
    ...
    {"id": "dapp-auction-house",
     "name": "Auction House",
     "photo-path": "contacts://auction-house",
     "dapp?": true,
     "dapp-url": "http://auctionhouse.dappbench.com"},
    ...
]
```

Is it fine?

File: src/status_im/chat/sign_up.cljs [hunk 1]
```diff
@@ -189,4 +186,7 @@
   {:whisper-identity console-chat-id
    :name             (s/capitalize console-chat-id)
    :photo-path       console-chat-id
-   :dapp?            true})
+   :dapp?            true
+   ; todo remove/change dapp config fot console
+   :dapp-url         "http://localhost:8185/resources"
```
Comment (rasom) [inline src/status_im/chat/sign_up.cljs]: It is not used at the moment, we discussed once that dapps will provide some url for js that will use our API to add command/requests

**Pull request merged**